src/components/Swap/Toolbar/Caption.tsx

```diff
@@ -68,57 +89,62 @@ export function MissingInputs() {
   return <Caption icon={Info} caption={<Trans>Enter an amount</Trans>} />
 }
 
-export function LoadingTrade() {
+export function LoadingTrade({ gasUseEstimateUSD }: GasEstimateProp) {
   return (
-    <Caption
-      icon={InlineSpinner}
-      caption={
-        <Loading>
-          <Trans>Fetching best price…</Trans>
-        </Loading>
-      }
-    />
+    <>
+      <Caption
+        icon={InlineSpinner}
+        caption={
+          <Loading>
+            <Trans>Fetching best price…</Trans>
+          </Loading>
+        }
+      />
+      <GasEstimate gasUseEstimateUSD={gasUseEstimateUSD} />
+    </>
   )
 }
 
-export function WrapCurrency({ inputCurrency, outputCurrency }: { inputCurrency: Currency; outputCurrency: Currency }) {
+interface WrapCurrencyProps {
+  inputCurrency: Currency
+  outputCurrency: Currency
+}
+
+export function WrapCurrency({
+  inputCurrency,
+  outputCurrency,
+  gasUseEstimateUSD,
+}: WrapCurrencyProps & GasEstimateProp) {
   const Text = useCallback(
     () => (
       <Trans>
-        Convert {inputCurrency.symbol} to {outputCurrency.symbol}
+        Convert {inputCurrency.symbol} to {outputCurrency.symbol} with no slippage
       </Trans>
     ),
     [inputCurrency.symbol, outputCurrency.symbol]
   )
 
-  return <Caption icon={Info} caption={<Text />} />
+  return (
+    <>
+      <Caption icon={Info} caption={<Text />} />
+      <GasEstimate gasUseEstimateUSD={gasUseEstimateUSD} />
+    </>
+  )
 }
 
-export function Trade({
-  trade,
-  outputUSDC,
-  impact,
-}: {
+export interface TradeProps {
   trade: InterfaceTrade
   outputUSDC?: CurrencyAmount<Currency>
   impact?: PriceImpact
-}) {
+}
+
+export function Trade({ trade, outputUSDC, impact, gasUseEstimateUSD }: TradeProps & GasEstimateProp) {
   return (
     <>
-      <Tooltip placement="bottom" icon={LargeIcon} iconProps={{ icon: impact?.warning ? AlertTriangle : Info }}>
-        <Column gap={0.75}>
-          {impact?.warning && (
-            <>
-              <ThemedText.Caption>
-                The output amount is estimated at {impact.toString()} less than the input amount due to impact
-              </ThemedText.Caption>
-              <Rule />
-            </>
-          )}
-          <RoutingDiagram trade={trade} />
-        </Column>
-      </Tooltip>
-      <Price trade={trade} outputUSDC={outputUSDC} />
+      <CaptionRow>
+        <Price trade={trade} outputUSDC={outputUSDC} />
```

Comment:
You've gutted the routing diagram without replacing it anywhere. Is that to be done in a follow-up PR? Right now it is just dead codepaths.

Reply:
yeah, i removed it intentionally - i figured with releases disabled this wouldn't be a problem until we add it back in the new location.